### PR TITLE
Double block execution prevention

### PR DESCRIPTION
**Pull request by m4ksio**
Fixes, what I believe was ultimate reason for flaky tests failing with `ComputeBlock` mock.
Previous change fixed possible block execution in real life but still made it possible to fail.
This fixes condition which are highly unlikely to happen in normal operation, but happens often in tests. In both cases it should help.

### CODE REVIEW

**Review by zhangchiqing — APPROVED** [review]

**Review by ramtinms — APPROVED** [review]
LGTM

**Comment by zhangchiqing** [comment]
Is it good to merge? I'm running into the flakey tests as well, hope this PR could resolve it

**Comment by m4ksio** [comment]
> Is it good to merge? I'm running into the flakey tests as well, hope this PR could resolve it

It fixes one flaky merge, but cases another. Its probably less frequent though, but @durkmurder volunteer to look into this.
But I guess it would be safe to merge it even if it just reduces the frequency of failing test

**Comment by zhangchiqing** [comment]
Maybe we could merge this and issue up another case

**Comment by m4ksio** [comment]
> Maybe we could merge this and issue up another case

Merged it, looks like tests run again without problem so there is hope

File: engine/execution/ingestion/engine.go [hunk 1]
```diff
@@ -804,6 +811,8 @@ func (e *Engine) handleCollection(originID flow.Identifier, collection *flow.Col
 				// the collection id matches with the CollectionID from the collection guarantee
 				completeCollection.Transactions = collection.Transactions
 
+				fmt.Printf("handled collection\n")
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine.go:814]: 👀

**Pull request merged**